### ISSUE TITLE
MemberNameEqualsClassName false positive? on built-in method whose name cannot change

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#4156]
## Expected Behavior of the rule
```kt
class Main {
	// lots of stuff here omitted for brevity
	companion object {

		@JvmStatic
		fun main(vararg args: String) {
		}
	}
}
```

## Context
For JVM compatibility it often happens that the main function is put into a companion object. And maybe it happens that the entry class of an app is also called Main. This makes `java foo.bar.Main` work, rather than `java foo.bar.MainKt`.

**Comment by BraisGabin** [comment]
What do you think about this?

```kotlin
@file:JvmName("Main")

fun main(vararg args: String) {
}
```

This way you should get `java.bar.Main`.

To me this is an edge case where adding a `@Suppress` should be the way to go.

**Comment by TWiStErRob** [comment]
Can't do that, because `// lots of stuff here omitted for brevity`. I'm using `class Main @Inject constructor(` which I can't do with a JvmName. It's fine to suppress for me, just wanted to raise. Are there may be other cases? Or probably we should be always more specific, for example when implementing `Comparator` we always do `FooComparator`.

**Comment by BraisGabin** [comment]
Yes, I understand that your cade wasn't that easy. But as I said I think that this is the kind of places where a `@Suppress` have a lot of sense. I don't know any other case but if there are more we could make somethign to fix it, for sure.

>  just wanted to raise

And that's great! Keep them flowing :). This is the way for detekt to improve.

**Comment by BraisGabin** [comment]
I was thinking... other option is to crete a new suppressor (I'm talking about #4068) to suppress classes by name. So in this case in the configuration of this rule you could add:

```yaml
MemberNameEqualsClassName:
  ...
  ignoreClassNamed:
    - 'Main'
```

It seems like a safe ignore that wouldn't hide too other errors and I was thinking already about implementing this "generic" suppressor because there are other rules that would use it too. What do you think? Would it help for your use case?
